lib/find-vs2017.js

```diff
@@ -0,0 +1,47 @@
+var log = require('npmlog')
+  , execSync = require('child_process').execSync
+  , path = require('path')
+
+var hasCache = false
+  , cache = null
+
+function findVS2017() {
+  if (hasCache)
+    return cache
+
+  hasCache = true
+
+  var ps = 'PowerShell -ExecutionPolicy Unrestricted -Command '
+  var csFile = path.join(__dirname, 'Find-VS2017.cs')
+  var psQuery = ps + '"&{Add-Type -Path \'' + csFile + '\'; [VisualStudioConfiguration.Main]::Query()}" 2>&1'
+
+  var vsSetup
+  try {
+    var vsSetupRaw = execSync(psQuery, { encoding: 'utf8' })
```

Comment:
Policy decision: do we still support v0.10?  If yes, you can't use execSync(), it was added in v0.12.

That aside, is there a reason to use execSync() instead of spawnSync()?  If `__dirname` has spaces in it, execSync() will probably fail.

Lastly, can powershell always be expected to be on the path?